src/Swarm/Language/Typecheck.hs

```diff
@@ -99,11 +100,18 @@ runInfer ctx =
 --   an 'UnboundVar' error if it is not found, or opening its
 --   associated 'UPolytype' with fresh unification variables via
 --   'instantiate'.
-lookup :: SrcLoc -> Var -> Infer UType
+lookup :: SrcLoc -> Var -> TC UType
 lookup loc x = do
   ctx <- ask
   maybe (throwError $ UnboundVar loc x) instantiate (Ctx.lookup x ctx)
 
+-- | Add a source location to a type error and re-throw it.
+addLocToTypeErr :: Syntax' ty -> TypeErr -> TC a
+addLocToTypeErr s te = case te of
+  UnifyErr NoLoc a b -> throwError $ UnifyErr (s ^. sLoc) a b
```

Comment:
can `throwError` be moved outside of the `case` expression?

Reply:
Good call, but I actually want to completely rewrite `addLocToTypeErr` (it should actually have a type like `SrcLoc -> TC a -> TC a`, and maybe a different name to go with it; it should handle all the different kinds of type errors, to facilitate which we should factor the `SrcLoc` out so it's not duplicated in every constructor, etc.) so I'd prefer to just wait and do all the refactoring at once.